libraries/AP_TemperatureSensor/TSYS01.h

```diff
@@ -16,7 +16,13 @@ class TSYS01 {
 
     bool init(void);
     float temperature(void) { return _temperature; } // temperature in degrees C
-    bool healthy(void) { return _healthy; } // do we have a valid temperature reading?
+    bool healthy(void) { // do we have a valid temperature reading?
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+        _temperature = 42.42;
```

Comment:
It would be better to set this on constructor, with the same macro guard; that way it doesn't depend on `healthy` being called.

Reply:
It's reset by the timer function, so setting it once there isn't an option.

Also, I just killed the constructor ;-)

Reply:
I hope the timer isn't running in SITL, otherwise something is really wrong! I keep my comment, setting in constructor is nicer 🙂